src/ecdsa/keys.py

```diff
@@ -1168,3 +1180,25 @@ def sign_number(self, number, entropy=None, k=None):
         assert 1 <= _k < order
         sig = self.privkey.sign(number, _k)
         return sig.r, sig.s
+
+    def ecdh_get_shared_secret(self, remote_public_key):
+        """
+        Elliptic-curve Diffie-Hellman (ECDH). A key agreement protocol.
+        Allows two parties, each having an elliptic-curve public-private key 
+        pair, to establish a shared secret over an insecure channel
+                
+        :param remote_public_key: public key of remote party (VerifyingKey)
+        :return: shared secret
+        """""
+
+        if self.curve.curve != remote_public_key.curve.curve:
+            raise InvalidPublicKeyCurveError
+
+        # shared secret = PUBKEYtheirs * PRIVATEKEYours
+        result = remote_public_key.pubkey.point * self.privkey.secret_multiplier
+        if result == ellipticcurve.INFINITY:
```

Comment:
we can test if this exception is raised by setting the `secret_multiplier` to either the value of order or 0 in the test case